python/whylogs/api/logger/__init__.py

```diff
@@ -20,6 +23,92 @@ def log(
     return TransientLogger(schema=schema).log(obj, pandas=pandas, row=row)
 
 
+def log_classification_metrics(
+    targets,
+    predictions,
+    scores=None,
+    target_field=None,
+    prediction_field=None,
+    score_field=None,
+) -> ProfileResultSet:
+    """
+    Function to track metrics based on validation data.
+    user may also pass the associated attribute names associated with
+    target, prediction, and/or score.
+    Parameters
+    ----------
+    targets : List[Union[str, bool, float, int]]
+        actual validated values
+    predictions : List[Union[str, bool, float, int]]
+        inferred/predicted values
+    scores : List[float], optional
+        assocaited scores for each inferred, all values set to 1 if not
+        passed
+    target_field : str, optional
+        Description
+    prediction_field : str, optional
+        Description
+    score_field : str, optional
+        Description
+    target_field : str, optional
+    prediction_field : str, optional
+    score_field : str, optional
+    """
+    model_performance_metrics = ModelPerformanceMetrics()
+    model_performance_metrics.compute_confusion_matrix(
+        predictions=predictions,
+        targets=targets,
+        scores=scores,
+        target_field=target_field,
+        prediction_field=prediction_field,
+        score_field=score_field,
+    )
+    profile = DatasetProfile()
+    profile.add_model_performance_metric(model_performance_metrics)
+    return ProfileResultSet(profile)
+
+
+def log_regression_metrics(
+    targets,
+    predictions,
+    target_field=None,
+    prediction_field=None,
```

Comment:
Same as described above, but for regression.

Reply:
done.